pygmt/tests/test_nearneighbor.py

```diff
@@ -0,0 +1,116 @@
+"""
+Tests for nearneighbor.
+"""
+import os
+
+import numpy.testing as npt
+import pytest
+import xarray as xr
+from pygmt import nearneighbor, which
+from pygmt.datasets import load_sample_bathymetry
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, data_kind
+
+
+@pytest.fixture(scope="module", name="ship_data")
+def fixture_ship_data():
+    """
+    Load the data from the sample bathymetry dataset.
+    """
+    return load_sample_bathymetry()
+
+
+def test_nearneighbor_input_file():
+    """
+    Run nearneighbor by passing in a filename.
+    """
+    fname = which("@tut_ship.xyz", download="c")
+    output = nearneighbor(
+        data=fname, spacing="5m", region=[245, 255, 20, 30], search_radius="10m"
+    )
+    assert isinstance(output, xr.DataArray)
+    assert output.gmt.registration == 0  # Gridline registration
+    assert output.gmt.gtype == 0  # Cartesian type
+    assert output.shape == (121, 121)
+    npt.assert_allclose(output.mean(), -2378.2385)
+    return output
```

Comment:
I know this matches what is used in `test_surface,py`, but there is not a similar test for other modules using remote data, such as `test_blockmedian.py` or `test_grdtrack.py`. Should we be implementing tests like this across the board that retrieve the input files directly instead of using the built-in data modules?

Reply:
Does this need implementing in this PR? There is a [blockmedian test](https://github.com/GenericMappingTools/pygmt/blob/bb618b7dbdaa0a44d9432ae34e90d6d93c0bb5a4/pygmt/tests/test_blockmedian.py#L91) which does not use `which`. Do you think this nearneighbor test should pass the filename directly, instead of using `which`?

Reply:
> Do you think this nearneighbor test should pass the filename directly, instead of using `which`?

Yes, I would recommend re-writing the test without using `which`. Unit tests should focus on testing a single 'unit' (e.g. the `nearneighbor` function) only whenever possible.

Reply:
I agree that the test should avoid using `which` if it can just be `data="@tut_ship.xyz"`. That's the format that I"m using in #1418.